src/module_wrap.cc

```diff
@@ -375,7 +375,13 @@ void ModuleWrap::Evaluate(const FunctionCallbackInfo<Value>& args) {
     return;
   }
 
-  args.GetReturnValue().Set(result.ToLocalChecked());
+  // If TLA is enabled, `result` is the evaluation's promise.
+  // Otherwise, `result` is the last evaluated value of the module,
+  // which could be a promise, which would result in it being incorrectly
+  // unwrapped when the higher level code awaits the evaluation.
+  if (env->isolate_data()->options()->experimental_top_level_await) {
```

Comment:
Should this maybe still be the v8 flag check? In case someone sets `--harmony-top-level-await`.

Reply:
`--harmony-top-level-await` implies `--experimental-top-level-await` and vice versa.